apps/managedidentity/managedidentity_test.go

```diff
@@ -180,13 +189,35 @@ func Test_SystemAssigned_Returns_Token_Success(t *testing.T) {
 					t.Fatalf("resource objectid is incorrect, wanted %s got %s", i.value(), query.Get(miQueryParameterObjectId))
 				}
 			}
-			if err != nil {
-				t.Fatal(err)
+			if result.Metadata.TokenSource != base.IdentityProvider {
+				t.Fatalf("wanted Indenity tokensource, got %d", result.Metadata.TokenSource)
 			}
 			if result.AccessToken != token {
 				t.Fatalf("wanted %q, got %q", token, result.AccessToken)
 			}
-
+			storage.FakeValidate = func(storage.AccessToken) error {
```

Comment:
Can we have some tests that don't do so much mocking, but instead rely on injecting HTTP responses ?

I think we want to test: 

- create MI client with { SAMI, UAMI by client_Id, UAMI by obj_id, UAMI by res_id } 
- get token, assert it's from Identity Provider
- get token again, assert it's from cache
- create MI2 client, get token, assert it's from cache.

Reply:
That is exactly what we are doing, We are injecting and http response 

- we are checking weather the http response was called.
- we are checking if the URL that it requested was correct.
- we also check if the first token is from identity provider
- we call `AcquireToken` second time and check if its from cache.
- We then create a new Managed identity of same type and check again if its called from cache.

The ` FakeValidate` we see here is that function that is used when we try to retrieve it from the cache, in reading process the storage checks if the token is valid or not. As, we have a fake token we don't that fails. To overcome that we provide a function that overrides that validation processes while testing. In the end we also check that was the validate function was called or not. This `FakeValidated` is what was used to check in other API's we have.

Reply:
Removed the `fakeValidation` removed form test.